### DISCUSSION TITLE
[Useful scripts] Pull arbitrary GitHub pull request

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#11139] [Show and tell]
There's nothing new, it's not difficult to [pull an arbitrary pull-request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#modifying-an-active-pull-request-locally) from GitHub, I don't do it regularly enough to memorise the command. 

Here's a script to slightly automate it:

```powershell
[string[]]$remotes = $(git remote);

if ($remotes.Count -gt 1) {
    $remotesList = $remotes -join "','"
    $remote = Read-Host -Prompt "GitHub remote (available '$remotesList')"
}
else {
    $remote = $remotes[0]
}

if ([string]::IsNullOrWhiteSpace($remote) -or $remotes -notcontains $remote) {
    Write-Host "Invalid remote name" -ForegroundColor Red
    exit -1
}

$pullId = 0;
$pullIdValue = Read-Host -Prompt "GitHub pull-request ID"
if ([string]::IsNullOrWhiteSpace($pullIdValue) -or ![int]::TryParse($pullIdValue, [ref] $pullId)) {
    Write-Host "Invalid pull-request ID" -ForegroundColor Red
    exit -1
}

git fetch $remote pull/$pullId/head:$remote-pull-$pullId

```
![image](https://github.com/dotnet/extensions/assets/4403806/e06f908b-f9ee-402e-a874-d577f5eba0f1)

Of course, I wanted more automation for the script, but I couldn't achieve it with the current implementation of the scripts subsystem.

**Comment by vbjay** [comment]
1.  ![image](https://github.com/gitextensions/gitextensions/assets/3629489/6b086769-c3bc-4215-97b4-5bab1ae8e5d7)
1.  ![image](https://github.com/gitextensions/gitextensions/assets/3629489/f63b2a53-2c5b-4f49-b3f6-60b16f4a5703)
1.  ![image](https://github.com/gitextensions/gitextensions/assets/3629489/09361b17-c304-4e7e-8350-ca9cef232cb7)

Note the buttons on the right.  Just adding the gui way for the people that like the gui vs script.

**Reply by RussKie** [reply]
Thank you. I've been avoiding the plugin because it hurts my aesthetic feelings :) One day I may get to fix the UX of this form.

**Comment by mstv** [comment]
![image](https://github.com/gitextensions/gitextensions/assets/36601201/1330a49f-d3ec-4dfa-beb2-289f9ec6eba0)

Just enter the PR number for the first script and get a branch with the same name.
The second script fetches updates.

**Reply by RussKie** [reply]
Nice! For some reason I couldn't get "userinput" to work....
